src/averageAges.js

```diff
@@ -37,7 +42,13 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const mothers = people.filter(el => el.mother).map(el => el.mother);
+  const women = withChildren ? people.filter(el => mothers.includes(el.name))
+    : people.filter(el => el.sex === 'f');
```

Comment:
Lots of unnecessary operations here.
Try to solve it using just one `filter` usage.
Hint: you can also use `some` method to check if a woman has any children.